core/src/main/java/io/grpc/internal/DelayedClientTransport2.java

```diff
@@ -85,13 +87,50 @@
   @Nullable
   private SubchannelPicker lastPicker;
 
-  DelayedClientTransport2(Executor streamCreationExecutor) {
-    this.streamCreationExecutor = streamCreationExecutor;
+  @GuardedBy("lock")
+  private long lastPickerVersion;
```

Comment:
The case I'm probably missing is how the LB behaves when the channel is IDLE and the races for RPCs involved. Yeah, it looks like this is actually necessary and not just a nice-to-have.